codecov_auth/tests/unit/test_repo_authentication.py

```diff
@@ -554,3 +555,130 @@ def test_tokenless_success(
         else:
             with pytest.raises(exceptions.AuthenticationFailed):
                 res = authentication.authenticate(request)
+
+
+class TestUploadTokenRequiredAuthenticationCheck(object):
+    def test_token_not_required_bad_path(self):
+        request = APIRequestFactory().post(
+            "/endpoint",
+            headers={},
+        )
+        authentication = UploadTokenRequiredAuthenticationCheck()
+        res = authentication.authenticate(request)
+        assert res is None
+
+    def test_token_not_required_bad_service(self):
+        request = APIRequestFactory().post(
+            "/upload/other/owner::::repo/commits/commit_sha/reports/report_code/uploads",
+            headers={"X-Tokenless": "user-name/repo-forked", "X-Tokenless-PR": "15"},
+        )
+        authentication = UploadTokenRequiredAuthenticationCheck()
+        res = authentication.authenticate(request)
+        assert res is None
+
+    def test_token_not_required_unknown_repository(self, db):
+        an_owner = OwnerFactory(upload_token_required_for_public_repos=False)
+        # their repo
+        RepositoryFactory(author=an_owner, private=False)
+        request_uri = f"/upload/github/{an_owner.username}::::bad/commits"
+        request = APIRequestFactory().post(
+            request_uri, {"branch": "branch"}, format="json"
+        )
+        authentication = UploadTokenRequiredAuthenticationCheck()
+        res = authentication.authenticate(request)
+        assert res is None
+
+    def test_token_not_required_unknown_owner(self, db):
+        repo = RepositoryFactory(
+            private=False, author__upload_token_required_for_public_repos=False
+        )
+        request_uri = f"/upload/github/bad::::{repo.name}/commits"
+        request = APIRequestFactory().post(
+            request_uri, {"branch": "branch"}, format="json"
+        )
+        authentication = UploadTokenRequiredAuthenticationCheck()
+        res = authentication.authenticate(request)
+        assert res is None
+
+    @pytest.mark.parametrize("request_uri,repo_slug,commitid", valid_params_to_test)
+    @pytest.mark.parametrize("private", [False, True])
+    @pytest.mark.parametrize("token_required", [False, True])
+    def test_token_not_required_matches_paths(
+        self, request_uri, repo_slug, commitid, private, token_required, db
+    ):
+        author_name, repo_name = repo_slug.split("/")
+        # Doing this because of ATS.
+        # For pytest '::' is the divider between a test class and a test function.
+        # There's a non-zero chance that the full test name would be mis-interpreted by pytest
+        # And these tests would never work with ATS.
+        # Sadly there's no way to escape the '::' sequence
+        request_uri = request_uri.replace("SEPARATOR", "::::")
```

Comment:
[nit] Ah this is no longer necessary.
Well 1 because we're not using ATS anymore, so who cares.
But 2 because @michelletran-codecov showed me that parametrized tests [can have an id](https://docs.pytest.org/en/stable/example/parametrize.html#different-options-for-test-ids).

So we can just add custom ids to the tests and we don't have to leave that SEPARATOR there.

I'm personally favor `pytest.param` with `id=` keyword argument